docs/javascript/v2/guides/javascript-quickstart.mdx

```diff
@@ -310,89 +414,107 @@ muteVid.addEventListener("click", () => {
+    bottom: 0;
+    width: 100%;
+    padding: 15px;
+    justify-content: center;
+    z-index: 10;
+  }
+
+  .control-bar > *:not(:first-child) {
+    margin-left: 8px;
+  }
 
-If you try it out now, you may notice that turning off the video makes the tile go blank but turning it on again doesn't bring it back.
-This is due to a change in video track requiring a call to attach again. To fix it let's
-call our `renderPeers` function whenever we change the state of the local video.
+  .btn-control {
+    font-size: 12px;
+    text-transform: uppercase;
+    letter-spacing: 1px;
+    border: 2px solid #37474f;
+    width: 64px;
+    height: 64px;
+    border-radius: 50%;
+    text-align: center;
+    background-color: #607d8b;
+    box-shadow: 0 0 10px rgba(0, 0, 0, 0.4);
+    color: white;
+  }
+```
 
-```jsx{8}:script.js section=MutingUnmutingLocalTracks sectionIndex=2
-muteVid.addEventListener("click", () => {
-  const videoEnabled = hmsStore.getState(selectIsLocalVideoEnabled);
+</Tab>
 
-  hmsActions.setLocalVideoEnabled(!videoEnabled);
+### Changing UI based on connection state
 
-  muteVid.textContent = !videoEnabled ? "Hide" : "Unhide";
+Right now, our join form will show up even after we have joined the room. We need a way to know the connection state of the room and hide the form after we've connected.
 
-  renderPeers(hmsStore.getState(selectPeers));
-});
+We can do this by subscribing to the store with the [selectIsConnectedToRoom](/api-reference/javascript/v2/home/content#select-is-connected-to-room) selector from `hmsStore`.
+
+```jsx:index.js section=ChangingUIBasedOnConnectionState sectionIndex=1
+// Showing the required elements on connection/disconnection
+function onConnection(isConnected) {
+  if (isConnected) {
+    form.classList.add("hide");
+    conference.classList.remove("hide");
+    leaveBtn.classList.remove("hide");
+    controls.classList.remove("hide");
+  } else {
+    form.classList.remove("hide");
+    conference.classList.add("hide");
+    leaveBtn.classList.add("hide");
+    controls.classList.add("hide");
+  }
+}
+
+// listen to the connection state
+hmsStore.subscribe(onConnection, selectIsConnectedToRoom);
 ```
 
-![Mute/Unmute](/guides/mute-unmute.gif)
+## Test the app
 
-## Mirroring local video
+### Start the ParcelJS development server
 
-Each `peer` object has an `isLocal` property that indicates whether the peer is local or remote.
-When looping over the peer list we can use this property to decide whether to mirror the video or not.
+To test the application, run the application using:
 
-<Tabs id="mirror" items={['JS', 'CSS']} />
+-   using Yarn `yarn parcel ParcelJS-test/index.html` or
+-   when using npm run `npx parcel index.html`.
 
-<Tab id="mirror-0" >
+This will start the app in a the Parcel built in development server, which will automatically rebuild your app as you make changes. Now open http://localhost:1234/ in your browser to test the video conferencing you created above.
```

Comment:
Let's use 3000 port, this para seems to be gramatucally off.

Reply:
fixed it